benchmarks/benchmark_shared.c

```diff
@@ -129,6 +129,9 @@ int bandwidth(void)
 	if (ret)
 		return ret;
 
+	struct fi_context *ctx_arr = 
+		calloc(opts.window_size, sizeof(struct fi_context));
+
```

Comment:
Please declare variables at the top of the function, to make them easier to find.  Also add a check for allocation failure.